pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -151,7 +157,10 @@ public static Expression getLiteralExpression(byte[] value) {
   }
 
   public static Expression getLiteralExpression(Object object) {
-    if (object instanceof Integer || object instanceof Long) {
+    if (object instanceof Integer) {
```

Comment:
I feel it is not needed to introduce int literal here. `long`, `double`, `string` should be enough to represent all literals

Reply:
I had made this change to make `CalciteSqlCompilerTest` pass, but it seems like those test cases also pass by using getIntValue() instead of getLongValue() so reverted this change. But I am not seeing much of a benefit in overloading long to represent both int and long. This can be confusing specially since int and long are different and can also cause correctness issues.

Reply:
Reverting this change will cause QueryOptimizerTest cases (testFlattenAndOrFilter, testMergeEqInFilter) to fail, more specifically the assert on line 107 and 178.

Reply:
I don't quite follow here. The test failure should not be related to this change but the new extra `NumericalFilterOptimizer`. It is okay to fix the test, but we should not modify the production code because of how tests are written.

Reply:
The assert on Line 107 in QueryOptimizerTest.java is:

```assertEquals(children.get(0), getEqFilterExpression("int", 4));```

where `children.get(0)` is:
```
Expression(type:FUNCTION, functionCall:Function(operator:EQUALS,
   operands:[
         Expression(type:IDENTIFIER, identifier:Identifier(name:int)),
         Expression(type:LITERAL, literal:<Literal intValue:4>)
   ]
))
```
The second operand of the expression above is an INT literal with value 4 because it got type casted to match the type of the column which is INT (INT literal matching INT column).

To make the assertEquals statement pass, `getEqFilterExpression("int", 4)` should return an INT literal; however, currently it is returning a LONG literal. To make `getEqFilterExpression("int", 4)` return INT literal, we need to modify the getLiteralExpression function as has been done above.

Reply:
For the existing behavior, we use Long literal to store all integral values, and use Double literal type to store all floating-point values. The optimizer should follow this rule, and use `RequestUtils.getLiteralExpression()` to create literal expressions. If both of them use the same method to create the expression, the test should pass

Reply:
We need to maintain reference to original expression (otherwise it will destroy the expression tree structure), so can't use `RequestUtils.getLiteralExpression()` as it will return a new expression object, but I have modified the `expression.getLiteral()` in place to use `long` and `double`.

As a side note, it seems like `RequestUtils.getLiteralExpression(int value)` will need to be fixed too (but don't want to touch it in this PR).

Reply:
To replace the rhs, you can simply set the second operands to the new rhs expression. In-place modification is also fine, but not as clean as creating a new one.